pandas/tests/base/test_ops.py

```diff
@@ -137,227 +137,233 @@ def setup_method(self, method):
+            assert not result.iat[1]
 
-    def test_ndarray_compat_properties(self):
+    def test_ndarray_compat_properties(self, index_or_series_obj):
+        o = index_or_series_obj
 
-        for o in self.objs:
-            # Check that we work.
-            for p in ["shape", "dtype", "T", "nbytes"]:
-                assert getattr(o, p, None) is not None
+        # Check that we work.
+        for p in ["shape", "dtype", "T", "nbytes"]:
+            assert getattr(o, p, None) is not None
 
-            # deprecated properties
-            for p in ["flags", "strides", "itemsize", "base", "data"]:
-                assert not hasattr(o, p)
+        # deprecated properties
+        for p in ["flags", "strides", "itemsize", "base", "data"]:
+            assert not hasattr(o, p)
 
-            msg = "can only convert an array of size 1 to a Python scalar"
-            with pytest.raises(ValueError, match=msg):
-                o.item()  # len > 1
+        msg = "can only convert an array of size 1 to a Python scalar"
+        with pytest.raises(ValueError, match=msg):
+            o.item()  # len > 1
 
-            assert o.ndim == 1
-            assert o.size == len(o)
+        assert o.ndim == 1
+        assert o.size == len(o)
 
         assert Index([1]).item() == 1
         assert Series([1]).item() == 1
 
-    def test_value_counts_unique_nunique(self):
-        for orig in self.objs:
-            o = orig.copy()
-            klass = type(o)
-            values = o._values
-
-            if isinstance(values, Index):
-                # reset name not to affect latter process
-                values.name = None
+    def test_value_counts_unique_nunique(self, index_or_series_obj):
+        orig = index_or_series_obj
+        o = orig.copy()
+        klass = type(o)
+        values = o._values
 
-            # create repeated values, 'n'th element is repeated by n+1 times
-            # skip boolean, because it only has 2 values at most
-            if isinstance(o, Index) and o.is_boolean():
-                continue
-            elif isinstance(o, Index):
-                expected_index = Index(o[::-1])
-                expected_index.name = None
-                o = o.repeat(range(1, len(o) + 1))
-                o.name = "a"
-            else:
-                expected_index = Index(values[::-1])
-                idx = o.index.repeat(range(1, len(o) + 1))
-                # take-based repeat
-                indices = np.repeat(np.arange(len(o)), range(1, len(o) + 1))
-                rep = values.take(indices)
-                o = klass(rep, index=idx, name="a")
-
-            # check values has the same dtype as the original
-            assert o.dtype == orig.dtype
-
-            expected_s = Series(
-                range(10, 0, -1), index=expected_index, dtype="int64", name="a"
-            )
+        if orig.duplicated().any():
+            pytest.skip("test implementation breaks with duplicated values")
```

Comment:
Is this a bug?

Reply:
Not in the application code, but rather that the test isn't written properly. I wanted to focus on using the fixtures instead of rewriting the tests for now.
-> I'd leave fixing this to a follow-up PR if that's ok.

Reply:
@SaturnFromTitan you're very good about following up on things, but things slip through the cracks, so let's make sure this is an xfail with a helpful comment describing what needs to be fixed